bot/cogs/core/paginator.py

```diff
@@ -9,26 +8,57 @@
 from bot.villager_bot import VillagerBotCluster
 
 PAGE_EMBED_CALLABLE = Callable[[int], discord.Embed | Coroutine[Any, Any, discord.Embed]]
-LEFT_ARROW = "⬅️"
-RIGHT_ARROW = "➡️"
-SKIP_TO_LAST = "⏩"
-BACK_TO_FIRST = "⏪"
-NAV_EMOJIS = [BACK_TO_FIRST, LEFT_ARROW, RIGHT_ARROW, SKIP_TO_LAST]
+
+
+class PaginatorView(discord.ui.View):
+    def __init__(self, get_page, page_count, timeout):
+        super().__init__(timeout=timeout)
+        self._page = 0
+        self._pages = page_count
+        self._get_page = get_page
+
+    @discord.ui.button(emoji="⏪", style=discord.ButtonStyle.gray)
+    async def first(self, interaction: discord.Interaction, button: discord.ui.Button):
+        self._page = 0
+        await self.update_message(interaction)
+
+    @discord.ui.button(emoji="⬅️", style=discord.ButtonStyle.gray)
+    async def back(self, interaction: discord.Interaction, button: discord.ui.Button):
+        self._page -= 1
+        await self.update_message(interaction)
+
+    @discord.ui.button(emoji="➡️", style=discord.ButtonStyle.gray)
+    async def next(self, interaction: discord.Interaction, button: discord.ui.Button):
+        self._page += 1
+        await self.update_message(interaction)
+
+    @discord.ui.button(emoji="⏩", style=discord.ButtonStyle.gray)
+    async def last(self, interaction: discord.Interaction, button: discord.ui.Button):
+        self._page = self._pages - 1
+        await self.update_message(interaction)
+
+    async def update_message(self, interaction: discord.Interaction):
+        # disable buttons if necessary
+        self.children[0].disabled = self._page == 0
+        self.children[1].disabled = self._page == 0
+        self.children[2].disabled = self._page == self._pages - 1
+        self.children[3].disabled = self._page == self._pages - 1
+
+        # update message with new page/embed
+        embed = await self._get_page(self._page)
+        await interaction.response.edit_message(embed=embed, view=self)
+
+    async def on_timeout(self):
+        for child in self.children:
+            child.disabled = True
```

Comment:
Instead of disabling the buttons, would it be better to just remove them (if possible)?

Reply:
That's possible but personally I feel like its better if we just disable the buttons. You prefer removing them completely?

Reply:
I guess it's fine